vita3k/cpu/src/dynarmic_cpu.cpp

```diff
@@ -0,0 +1,345 @@
+        }
+
+        return ret;
+    }
+
+    uint32_t MemoryRead32(Dynarmic::A32::VAddr addr) override {
+        uint32_t ret = *Ptr<uint32_t>(addr).get(*parent->mem);
+
+        if (log_read) {
+            LOG_TRACE("Read uint32_t at address: 0x{:x}, val = 0x{:x}", addr, ret);
+        }
+
+        return ret;
+    }
+
+    uint64_t MemoryRead64(Dynarmic::A32::VAddr addr) override {
+        uint64_t ret = *Ptr<uint64_t>(addr).get(*parent->mem);
+
+        if (log_read) {
+            LOG_TRACE("Read uint64_t at address: 0x{:x}, val = 0x{:x}", addr, ret);
+        }
+
+        return ret;
+    }
+
+    void MemoryWrite8(Dynarmic::A32::VAddr addr, uint8_t value) override {
+        *Ptr<uint8_t>(addr).get(*parent->mem) = value;
+
+        if (log_write) {
+            LOG_TRACE("Write uint8_t at addr: 0x{:x}, val = 0x{:x}", addr, value);
+        }
+    }
+
+    void MemoryWrite16(Dynarmic::A32::VAddr addr, uint16_t value) override {
+        *Ptr<uint16_t>(addr).get(*parent->mem) = value;
+
+        if (log_write) {
+            LOG_TRACE("Write uint16_t at addr: 0x{:x}, val = 0x{:x}", addr, value);
+        }
+    }
+
+    void MemoryWrite32(Dynarmic::A32::VAddr addr, uint32_t value) override {
+        *Ptr<uint32_t>(addr).get(*parent->mem) = value;
+
+        if (log_write) {
+            LOG_TRACE("Write uint32_t at addr: 0x{:x}, val = 0x{:x}", addr, value);
+        }
+    }
+
+    void MemoryWrite64(Dynarmic::A32::VAddr addr, uint64_t value) override {
+        *Ptr<uint64_t>(addr).get(*parent->mem) = value;
+
+        if (log_write) {
+            LOG_TRACE("Write uint64_t at addr: 0x{:x}, val = 0x{:x}", addr, value);
+        }
+    }
+
+    void InterpreterFallback(Dynarmic::A32::VAddr addr, size_t num_insts) {
+        const bool thumb = cpu->is_thumb_mode();
+
+        if (thumb) {
+            addr &= 0xFFFFFFFE;
+        } else {
+            addr &= 0xFFFFFFFC;
+        }
+        static std::set<uint32_t> logged;
+        static std::mutex mu;
+        mu.lock();
+        if (logged.find(addr) == logged.end()) {
+            LOG_TRACE("Fallback at addr 0x{:X}, inst 0x{:X} ({})", addr, MemoryReadCode(addr), disassemble(*parent, addr, nullptr));
+            logged.insert(addr);
+        }
+        mu.unlock();
+        CPUContext context = cpu->save_context();
+        cpu->fallback.load_context(context);
+        std::uint64_t res = static_cast<std::uint64_t>(cpu->fallback.execute_instructions_no_check(static_cast<int>(num_insts)));
+        context = cpu->fallback.save_context();
+        cpu->load_context(context);
+
+        interpreted += num_insts;
+    }
+
+    void ExceptionRaised(uint32_t pc, Dynarmic::A32::Exception exception) override {
+        switch (exception) {
+        case Dynarmic::A32::Exception::Breakpoint:
+            return;
+        case Dynarmic::A32::Exception::UndefinedInstruction: {
+            break;
+        }
+        default:
```

Comment:
What is the difference? Return and break look different but do the same.

Reply:
Just remainder of print debugging. Thanks for point out.